tests/database.test.js

```diff
@@ -0,0 +1,139 @@
+const app = require('../server/app.js');
+const supertest = require("supertest");
+const http = require('http');
+const mongoose = require('mongoose');
+const { User, Course, SeatingLayout } = require('../server/dbSchemas/attendanceSchema.js');
+
+describe('Database Functionality', () => {
+
+    let request;
+    let server;
+
+    // Open database before running tests
+    beforeAll(async (done) => {
+        server = http.createServer(app);
+        server.listen();
+        request = supertest(server);
+        const db = mongoose.connection;
+        db.once('open', done);
+    });
+
+    // Close database & server when done
+    afterAll(async (done) => {
+        await mongoose.connection.close();
+        await server.close(done);
+    });
+
+    it('Should store user to database', async done => {
+        //Send request to register user
+        const res = await request.post('/api/register').send({
+            email: 'dummy@test.com',
+            name: 'dummy test',
+            password:'12345',
+            is_professor: true
+        });
+
+        //Search user in database by email
+        const user = await User.findOne({
+            email: 'dummy@test.com'
+        });
+
+        //Check that user has name, email, password, and is or is not a professor
+        expect(user.name).toBeTruthy()
+        expect(user.email).toBeTruthy()
+        expect(user.password).toBeTruthy()
+        expect(typeof(user.is_professor) === 'boolean').toBeTruthy();
+
+        //Delete test user after testing
+        response = await request.delete("/api/delete-user").send({
+            email: 'dummy@test.com'
+        });
+
+        done ()
+    });
+
+    xit('Should store course to database', async done => {
+
+        // Create a sample seating layout for this test
+        const seatingLayout = await request.post('/api/createSeatingLayout').send({
+            name: 'Sample layout',
+            capacity: 25,
+            dimensions: [ 5 , 5],
+            layout: [[5]], 
```

Comment:
Here remember that the layout is a 2d array representing each seat where each seat has a number associated with it which determines the type (0=locked, 1=open-access, 2=selected access, 3=extended access). Therefore the layout should look more like this:
```suggestion
            layout: [
                [2, 1, 1, 1, 0],
                [2, 1, 1, 1, 0],
                [2, 1, 1, 1, 0],
                [2, 1, 1, 1, 0],
                [2, 3, 3, 3, 0],
            ], 
```